tools/autograd/gen_trace_type.py

```diff
@@ -327,7 +327,10 @@ def emit_trace_body(f: NativeFunction) -> List[str]:
     dispatcher_exprs = dispatcher_sig.exprs()
 
     ret_and_arg_types = ', '.join([dispatcher_sig.returns_type()] + [a.type.cpp_type() for a in dispatcher_exprs])
-    redispatch_args = ', '.join(['op', 'c10::DispatchKey::Tracer'] + [a.expr for a in dispatcher_exprs])
+    # code-generated tracing kernels plumb and recompute dispatch keys directly through the kernel for performance.
+    # See Note [Plumbing Keys Through The Dispatcher] for details.
+    dispatch_key_set = 'ks & c10::after_tracing_keyset'
```

Comment:
nit: fwiw, I don't think pre-defining after_tracing_keyset is doing you much good, since FULL_AFTER is already a constexpr constructor, so the compiler will be able to figure it out either way.

Reply:
Same thing for `after_autograd_keyset` then? The only real difference is that `after_autograd_keyset` is also used in `VariableTypeManual.cpp`, not just codegen. But yeah they're both just syntactic sugar and have no runtime difference.

I left a comment[ further up](https://github.com/pytorch/pytorch/pull/49354#discussion_r560576934) in the PR about what to do about simplifying the redispatch API in general (both a potential avenue + me questioning whether it's really needed in this PR), so lmk if you agree/disagree.